### PR TITLE
[Debug] Debug toolbar resume button stuck in wrong state

### PR DESCRIPTION
**Pull request by matt-oster**
- Handle possible promise rejection from doSendRequest on unsuccessful messages
    
Signed-off-by: Matt Oster <matthew.oster@arm.com>

**Problem**

The resume button in the debug toolbar gets stuck in a disabled state

**Cause**

The updateBreakpoints method in debug-session.tsx doesn't handle promise rejection from doSendRequest in debug-session-connection.ts, which will happen anytime a debug adapter sends back an error response to a setBreakpoints request.

**Reproduction**

1. Have two separate projects in the workspace
2. Set break points in a file from each project
3. Debug with cortex-debug adapter

When the adapter handles breakpoints from a file that is not in the debugged binary, the MI is:
* 33^error,msg="No source file named /Users/matost01/mbed-workspace/mbed-os-example-blinky-2/main.cpp."

and the debug adapter sends back a message to doSendRequest like:
* { ..., success: false, body: { error: "..."}}

Since doSendRequest does a promise rejection on any message where success is false, the problem is triggered. The adapter isn't doing anything unreasonable in returning an unsuccessful message as it was in fact unable to handle the requested breakpoint. This scenario works fine in VS Code with the cortex-debug adapter with no UI impact for example.

**Fix**

I added a try/catch block to handle the promise rejection from doSendRequest and for all enabled breakpoints for the URI that received an unsuccessful message, I set verify and message.

### CODE REVIEW

**Comment by matt-oster** [comment]
@akosyakov @tolusha, would you mind having a look?

File: packages/debug/src/browser/debug-session.tsx [hunk 1]
```diff
@@ -512,13 +512,28 @@ export class DebugSession implements CompositeTreeElement {
...
+                    breakpoints: enabled.map(({ origin }) => origin.raw)
+                });
+                response.body.breakpoints.map((raw, index) => enabled[index].update({ raw }));
+            } catch (error) {
+                // handle adapters that send unsuccessful messages with error body for invalid breakpoints
+                enabled.forEach((brkPoint: DebugBreakpoint) => {
+                    const debugBreakpointData: Partial<DebugBreakpointData> = {
+                        raw: {
+                            verified: false,
+                            message: 'Breakpoint not valid for current debug session'
```
Comment (tolusha) [inline packages/debug/src/browser/debug-session.tsx]: It is better to grab message from `error` ? It seems to be more clear.

File: packages/debug/src/browser/debug-session.tsx [hunk 2]
```diff
@@ -512,13 +512,28 @@ export class DebugSession implements CompositeTreeElement {
...
+
+            try {
+                const response = await this.sendRequest('setBreakpoints', {
+                    source: source.raw,
+                    sourceModified,
+                    breakpoints: enabled.map(({ origin }) => origin.raw)
+                });
+                response.body.breakpoints.map((raw, index) => enabled[index].update({ raw }));
+            } catch (error) {
+                // handle adapters that send unsuccessful messages with error body for invalid breakpoints
```
Comment (akosyakov) [inline packages/debug/src/browser/debug-session.tsx]: if `error` is not `ResponseError`, we should log it with `console.error`.

**Comment by matt-oster** [comment]
Yes, VS code just carries on and does nothing special with the problem breakpoinst as they show as active even though they aren't valid. I chose to update the breakpoints so they would show as disabled in files outside of what is actually being debugged.

I used a generic message as it might make more sense in the UI than whatever might be coming from the debug adapters.

I don't mind changing any of it, that was just my rationalization for the chages I did make. So do you prefer going VS code way and do nothing when an error is retuned (except maybe log it)?

File: packages/debug/src/browser/debug-session.tsx [hunk 3]
```diff
@@ -512,13 +512,36 @@ export class DebugSession implements CompositeTreeElement {
...
+                const response = await this.sendRequest('setBreakpoints', {
+                    source: source.raw,
+                    sourceModified,
+                    breakpoints: enabled.map(({ origin }) => origin.raw)
+                });
+                response.body.breakpoints.map((raw, index) => enabled[index].update({ raw }));
+            } catch (error) {
+                // could be error or promise rejection of DebugProtocol.SetBreakpointsResponse
+                if (error instanceof Error) {
+                    console.error(`Error setting breakpoints: ${error.message}`);
```
Comment (akosyakov) [inline packages/debug/src/browser/debug-session.tsx]: `console.error('Error setting breakpoints:', error);` would be better

`error.message` does not contain stack trace information

**Pull request merged**